### PR TITLE
Fix Infinite Loops not happening issue and issue #12

### PR DESCRIPTION
**Pull request by shashikantx**
Comparison fails because `page` is of type `double` and `length` of type `int`.

also should fix issue #12

### CODE REVIEW

**Comment by moshiplus** [comment]
> [ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
> E/flutter ( 2218): 'package:flutter/src/widgets/page_view.dart': Failed assertion: line 96 pos 7: 'positions.isNotEmpty': PageController.page cannot be accessed before a PageView is built with it.
> E/flutter ( 2218): #0      _AssertionError._doThrowNew (dart:core/runtime/liberrors_patch.dart:37:39)
> E/flutter ( 2218): #1      _AssertionError._throwNew (dart:core/runtime/liberrors_patch.dart:33:5)
> E/flutter ( 2218): #2      PageController.page (package:flutter/src/widgets/page_view.dart)
> E/flutter ( 2218): #3      CarouselState.initState.<anonymous closure> (package:carousel_pro/src/carousel_pro.dart:113:24)
> E/flutter ( 2218): #4      _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
> E/flutter ( 2218): #5      _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
> E/flutter ( 2218): #6      _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:165:12)


still problem not solved

**Comment by shashikantx** [comment]
Yes. I cannot get a handle on it, though, I tried initiating those call on PageController after build, but it still persists.

**Comment by moshiplus** [comment]
i try to fix it

**Comment by shashikantx** [comment]
Finally fixed issue #12 ,

used [](https://docs.flutter.io/flutter/widgets/PageController/page.html)

File: lib/src/carousel_pro.dart [hunk 1]
```diff
@@ -107,7 +107,7 @@ class CarouselState extends State<Carousel> {
     
     if(widget.autoplay) {
       new Timer.periodic(widget.autoplayDuration, (_) {
-        if(_controller.page.toInt() == widget.images.length-1) {
+        if(_controller.page.round() == widget.images.length-1) {
```
Comment (deetu) [inline lib/src/carousel_pro.dart]: This is the correct solution to this issue.

**Pull request merged**